ffi/diplomat/src/logging.rs

```diff
@@ -11,14 +11,44 @@ pub mod ffi {
     pub struct ICU4XLogger;
 
     impl ICU4XLogger {
-        /// Initialize the logger from the `simple_logger` crate, which simply logs to
-        /// stdout. Returns `false` if there was already a logger set, or if logging has not been
+        /// Initialize the logger using `simple_logger`, or console.log/warn in WASM.
+        ///
+        /// Returns `false` if there was already a logger set, or if logging has not been
         /// compiled into the platform
         pub fn init_simple_logger() -> bool {
-            #[cfg(not(all(not(target_arch = "wasm32"), feature = "simple_logger")))]
-            unimplemented!("ICU4X binary not compiled with simple_logger support");
+            #[cfg(all(not(target_arch = "wasm32"), not(feature = "simple_logger")))]
+            return false;
```

Comment:
We used to fail loudly if this function was called in an environment without logging support. Now we fail quietly. Problem?

Reply:
Well the docs say we don't fail loudly

Reply:
Probably best to make this a linker error and update the docs